portfolio/src/main/java/com/google/sps/servlets/models/CommentReader.java

```diff
@@ -0,0 +1,101 @@
+package com.google.sps.servlets.models;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Transaction;
+import com.google.appengine.api.datastore.TransactionOptions;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+public abstract class CommentReader {
+  private final static Logger LOGGER = Logger.getLogger(CommentReader.class.getName());
+
+  public static List<Comment> getAll() {
+    Query.Filter publicOnlyFilter = new Query.FilterPredicate(
+            "status", Query.FilterOperator.EQUAL, Comment.Status.PUBLIC.name()
+    );
+    Query query = new Query("Comment")
+            .setFilter(publicOnlyFilter)
+            .addSort("timestamp", Query.SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<Comment> comments = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      Comment comment = Comment.fromEntity(entity);
+      comments.add(comment);
+    }
+    return comments;
+  }
+
+  public static String insertOne(String userId, RequestData requestData) {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Transaction transaction = datastore.beginTransaction();
+    String successResponse = "";
+    String failureResponse = "Failed to post comment. Please try again";
```

Comment:
just return Comment and throw an exception if it fails.  You can catch the exception in the caller.  Alternatively you could return null.  Either way the actual error message returned to the front-end (which the front-end depends on according to your JS code) should be in the actual servlet, not in this class that doesn't know who's calling it.